infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -380,23 +379,29 @@ private void waitMachines(
     }
   }
 
-  private void finishAllStartupTracesAsFailures(Throwable reason) {
-    machineStartupTraces
-        .entrySet()
-        .removeIf(
-            e -> {
-              finishSpanAsFailure(e.getValue(), reason.getMessage());
-              return true;
-            });
-  }
-
   /**
    * Returns a function, the result of which the completable stage that performs servers checks and
    * start of servers probes.
    */
   private Function<Void, CompletionStage<Void>> checkServers(
       List<CompletableFuture<?>> toCancelFutures, KubernetesMachineImpl machine) {
+
+    // Need to get active span here to allow use in returned function
+    final Span activeSpan = tracer == null ? null : tracer.activeSpan();
+
     return ignored -> {
+      final Span tracingSpan =
+          tracer == null
+              ? null
```

Comment:
This complicates code quite significantly. Can you move it to a separate method?

Reply:
Moved to `TracerUtil`